### DISCUSSION TITLE
Buying woes and PSRAM issues

### DISCUSSION THREAD

**Discussion by magicus** [jomjol/AI-on-the-edge-device#2199] [General]
I'm trying to build a smart meter using Aote. I bought a ESP32-CAM module off AliExpress, one that was listed as working in the "List of working Hardware" thread. However, after flashing it with Aote, I got the dreaded `PSRAM ID read error: 0xffffffff`.

I am prepared to order another board, but now I want to be super careful about ordering something that works. I just don't understand how to go about to do that. It all seems hit and miss, and the vendors are switching PSRAM module willy-nilly so someone ordering a successful component does not guarantee that the next one you want to order is OK.

Is the situation really that bad? Then it seems the best you can do is to order like 3 of the cheapest boards from different vendors, and hope that at least one of them will work. That seems so stupid. I am prepared to pay what it costs if I can be guaranteed that it will work, but I can't find a way to even do that.

Secondly, I wonder how this can be a problem, and that only AI-on-the-edge seem to be affected. Is it because no other projects out there really use the PSRAM, and do not notice that the memory is broken? 

Also, in my case, there were a demo firmware installed on the chip. I could connect to it just fine and see the camera image on a web page. Unfortunately, that firmware is now overwritten, and I did not check any logs. But how come that this worked? Can it work without using the PSRAM, or is there a configuration problem with Aote, rather than broken hardware?

**Comment by jomjol** [comment]
You need the PSRAM otherwise there is not enough memory to handle the CNNs.

**Reply by magicus** [reply]
Ok, I'm starting to understand the problem a bit now. Many other application do not use the PSRAM, which probably has led the problem with faulty PSRAMs to go undetected.

I tried digging deeper into ESP32, Espressif-IDF and PSRAM configuration, to see if it was "just" a misconfiguration. (I'm a hardware noob so this is all quite new to me). I had the suspicion that maybe the PSRAM itself were actually working, but the ESP board had a non-standard configuration, but I failed to make any progress along those lines.

And I have ordered two new ESPs from other vendors, hoping one of them will work... *sigh*

**Reply by caco3** [reply]
There are already discussions in here people say they replaced the PSRAM chip and then it worked...

Also see https://heise.de/-7535551

**Reply by magicus** [reply]
My German is a bit shaky, but if the connections are missing on the board then replacing the PSRAM won't help..? In any case, I'm totally worthless with a soldering iron, so I'll bet my luck on the incoming new ESP32s. 

As long as this situation persists, I wonder if the documentation should be made clearer for newcomers like me that it is currently a minefield and a lottery if you will get a working unit or not. I understood that buying proper hardware was important, but I thought that reading the specs, and following recommendations from other buyers would be enough to ensure a working unit.

Sorry if I'm sounding negative. I was just so disappointed after waiting weeks for the unit, and then it did not work. I am so eager to get this awesome project working!
